### ISSUE TITLE
[DOC] write complete docstrings for performance metrics classes

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#4303]
The docstrings of performance metrics classes in `performance_metrics.forecasting._classes` should be completed.

For each metric, the following should be done:

* a mathematical summary of the metric should be provided, including for the output of `evaluate_by_index`
* it should be ensured that the `multilevel` and `multioutput` arguments are included and documented.

[This PR](https://github.com/sktime/sktime/pull/4302) can be taken as a blueprint for improvements/changes to the docstrings.

If you would like to work on this, please indicate here the name of the metrics class, and whether you will work on math, parameters/arguments, or both.

**Comment by ramsarank** [comment]
Hello @fkiraly 

Just wanted to check if there are any pending metrics classes.
Thank you.

**Comment by fkiraly** [comment]
Yes! Examples are `MedianAbsoluteScaledError`, `MeianAbsoluteScaledErro`.

Would be great if you could help, @ramsarank!

**Comment by ramsarank** [comment]
Sure, I shall work on this. Thank you @fkiraly

**Comment by nkc-137** [comment]
Hello @fkiraly, 
I went through the `_classes` file and it looks like all the metric classes have doc strings and parameters already. But I saw that some metrics like `MeanSquaredError` and `MedianAbsoluteError` don't have mathemical formulae as the metric `MeanAbsoluteError` doc string has. Is this something I could potentially work on? 

Thanks!

**Comment by fkiraly** [comment]
> Is this something I could potentially work on?

Absolutely! Thanks!

Another, related, task is to implement efficient `_evaluate_by_index` methods: https://github.com/sktime/sktime/issues/4304

There's some synergy doing the two for one given metric at the same time, but of course it's not required to do so.

**Comment by sinemkilicdere** [comment]
> > Is this something I could potentially work on?
> 
> Absolutely! Thanks!
> 
> Another, related, task is to implement efficient `_evaluate_by_index` methods: [#4304](https://github.com/sktime/sktime/issues/4304)
> 
> There's some synergy doing the two for one given metric at the same time, but of course it's not required to do so.

I would love to work on _mlinex.py.
